api/moderation_auth.go

```diff
@@ -21,32 +71,66 @@ type authContext struct {
 // TODO: Allow a authOpts to be passed in with a description of attempted
 // action.
 func (a *AuthEngine) auth(ctx context.Context, req connect.AnyRequest) (*authContext, error) {
+	// Extract the token from the headers
 	authHeader := req.Header().Get("Authorization")
 	if authHeader == "" {
 		return nil, connect.NewError(connect.CodeUnauthenticated, fmt.Errorf("no token provided"))
 	}
-
 	authParts := strings.Split(authHeader, " ")
 	if len(authParts) != 2 {
 		return nil, connect.NewError(connect.CodeUnauthenticated, fmt.Errorf("malformed header"))
 	}
-
 	if authParts[0] != "Bearer" {
 		return nil, connect.NewError(connect.CodeUnauthenticated, fmt.Errorf("only Bearer auth supported"))
 	}
 
-	// Check the presented token is valid against the real bsky.
-	// This also lets us introspect information about the user - we can't just
-	// parse the JWT as they do not use public key signing for the JWT.
-	_, tokenDID, err := bluesky.ClientFromToken(ctx, a.PDSHost, authParts[1])
+	// Validate the token from the header
+	did, err := a.TokenValidator(ctx, authParts[1])
```

Comment:
Yeah - it's the same in the current implementation. It's not great but they don't sign their tokens with a pub/priv keypair so the only way to validate their token is to actually make a request with it 😓 

What I'll do eventually is add an exchange endpoint where they can swap a BlueSky token for a BFF token and we can encode their DID and maybe even their roles into that (if we want to avoid a db hit on each request - which seems probably more effort than its worth for such a low traffic API)

Reply:
That's a great idea - will address in a follow up PR